src/features/device-state/routes/state-patch-v2.ts

```diff
@@ -268,6 +269,22 @@ export const statePatchV2: RequestHandler = async (req, res) => {
 
 				updateFns.push(async (resinApiTx) => {
 					if (Object.keys(deviceBody).length > 0) {
+						// truncate for resilient legacy compatible device state patch so that supervisors don't fail
+						// to update b/c of length violation of 255 (SBVR SHORT TEXT type) for ip and mac address.
+						// sbvr-types does not export SHORT TEXT VARCHAR length 255 to import.
+						if (deviceBody.ip_address) {
+							deviceBody.ip_address = deviceBody.ip_address?.slice(
```

Comment:
since you are inside the `if`, I expect that the `?.` is not needed.
```suggestion
							deviceBody.ip_address = deviceBody.ip_address.slice(
```